### PR TITLE
chore: update otel to v0.13.0

### LINKED ISSUE #324: Update otel to latest version
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
- Update otel dependency to latest release 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Kubernetes version: (use `kubectl version`):

/good-first-issue

### PR DESCRIPTION
**Pull request by vladdy**
**What this PR does / why we need it**:
This updates the dependency to the latest release: 0.13.0

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #324

**Special notes for your reviewer**:
Since Prometheus exporter now uses a much simpler `pull.Controller` which does not have the `Stop()` method, I took a shortcut and just removed its call from `cmd/secrets-store-csi-driver/main.go`. Let me know if it would still be preferable to wrap the Prometheus' HTTP handler into `http.server` and export its `Shutdown()` method in a generic `Stop()`.

### CODE REVIEW

File: pkg/rotation/stats_reporter.go [hunk 1]
```diff
@@ -30,7 +29,7 @@ var (
 	rotatedKey                  = "rotated"
 	rotationReconcileTotal      metric.Int64Counter
 	rotationReconcileErrorTotal metric.Int64Counter
-	rotationReconcileDuration   metric.Float64Measure
+	rotationReconcileDuration   metric.Float64Counter
```
Comment (aramase) [inline pkg/rotation/stats_reporter.go:32]: This should be `Float64ValueRecorder` as it records duration

File: pkg/rotation/stats_reporter.go [hunk 2]
```diff
@@ -48,20 +47,20 @@ func newStatsReporter() StatsReporter {
 	meter := global.Meter("secretsstore")
 	rotationReconcileTotal = metric.Must(meter).NewInt64Counter("total_rotation_reconcile", metric.WithDescription("Total number of rotation reconciles"))
 	rotationReconcileErrorTotal = metric.Must(meter).NewInt64Counter("total_rotation_reconcile_error", metric.WithDescription("Total number of rotation reconciles with error"))
-	rotationReconcileDuration = metric.Must(meter).NewFloat64Measure("rotation_reconcile_duration_sec", metric.WithDescription("Distribution of how long it took to rotate secrets-store content for pods"))
+	rotationReconcileDuration = metric.Must(meter).NewFloat64Counter("rotation_reconcile_duration_sec", metric.WithDescription("Distribution of how long it took to rotate secrets-store content for pods"))
```
Comment (aramase) [inline pkg/rotation/stats_reporter.go:50]:
```suggestion
	rotationReconcileDuration = metric.Must(meter).NewFloat64ValueRecorder("rotation_reconcile_duration_sec", metric.WithDescription("Distribution of how long it took to rotate secrets-store content for pods"))
```

File: pkg/secrets-store/stats_reporter.go [hunk 3]
```diff
@@ -56,33 +55,33 @@ func NewStatsReporter() StatsReporter {
 	nodePublishErrorTotal = metric.Must(meter).NewInt64Counter("total_node_publish_error", metric.WithDescription("Total number of node publish calls with error"))
 	nodeUnPublishErrorTotal = metric.Must(meter).NewInt64Counter("total_node_unpublish_error", metric.WithDescription("Total number of node unpublish calls with error"))
 	syncK8sSecretTotal = metric.Must(meter).NewInt64Counter("total_sync_k8s_secret", metric.WithDescription("Total number of k8s secrets synced"))
-	syncK8sSecretDuration = metric.Must(meter).NewFloat64Measure("sync_k8s_secret_duration_sec", metric.WithDescription("Distribution of how long it took to sync k8s secret"))
+	syncK8sSecretDuration = metric.Must(meter).NewFloat64Counter("sync_k8s_secret_duration_sec", metric.WithDescription("Distribution of how long it took to sync k8s secret"))
```
Comment (aramase) [inline pkg/secrets-store/stats_reporter.go:58]:
```suggestion
	syncK8sSecretDuration = metric.Must(meter).NewFloat64ValueRecorder("sync_k8s_secret_duration_sec", metric.WithDescription("Distribution of how long it took to sync k8s secret"))
```

File: pkg/secrets-store/stats_reporter.go [hunk 4]
```diff
@@ -32,7 +31,7 @@ var (
 	nodePublishErrorTotal   metric.Int64Counter
 	nodeUnPublishErrorTotal metric.Int64Counter
 	syncK8sSecretTotal      metric.Int64Counter
-	syncK8sSecretDuration   metric.Float64Measure
+	syncK8sSecretDuration   metric.Float64Counter
```
Comment (aramase) [inline pkg/secrets-store/stats_reporter.go:34]:
```suggestion
	syncK8sSecretDuration   metric.Float64ValueRecorder
```

File: pkg/rotation/stats_reporter.go (hunk 1 above)
Comment (aramase) [inline pkg/rotation/stats_reporter.go:32]:
```suggestion
	rotationReconcileDuration   metric.Float64ValueRecorder
```

**Review by aramase — CHANGES_REQUESTED** [review]
@vladdy Thank you for the PR. Added few comments.

The duration needs to be of type `Float64ValueRecorder` so it can record the duration and add to the histogram bucket.

**Comment by vladdy** [comment]
@aramase, True, I missed that part. Thanks for catching!

**Review by aramase — COMMENTED** [review]
@vladdy Thank you for the PR  🎉 

/lgtm

**Pull request merged**